File: 65.valid-number.cpp
class Solution {
public:
    bool isNumber(string s) {
        // \s* space
        // \d+(\.\d*)?|\.\d+ float
        // [+-]?\\d+ integer
        regex rg(R"(\s*[+-]?(\d+(\.\d*)?|\.\d+)(e[+-]?\d+)?\s*)");
        return std::regex_match(s, rg);
    }
};

File: 65.valid-number.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isNumber = function(s) {
    // \s* space
    // \d+(\.\d*)?|\.\d+ float
    // [+-]?\\d+ integer
    let regex = /^\s*[+-]?(\d+(\.\d*)?|\.\d+)(e[+-]?\d+)?\s*$/;
    return regex.test(s);
};